### ISSUE TITLE
Tutorial fixes

### ISSUE THREAD

**Issue by jeremy-myers** [sandialabs/pyttb#228]
Pyttb Tutorial
# General
* Suggestion: re-order tutorial topics as
    * Tensors
    * Converting a Tensor to a Matrix and Vice Versa
    * Sparse Tensors
    * Kruskal Tensors
    * Alternating Poisson Regression for fitting CP to sparse count data
        * Suggestion: add parenthetical “(CP-APR)” to title
    * Tucker Tensors

# Kruskal Tensors
## Use tendiag or sptendiag to convert a ktensor to a ttensor
* Y is not defined in first example
## Use negative indexing and end for the last array index
* Could be interpreted that there is a keyword `end`

# Converting a Tensor to a Matrix and Vice Versa
- [ ] Documentation: descriptive documentation of `ttb.tenmat.from_tensor_type()`
- [ ] Suggestion: include definition of matriculation from Kolda & Bader (Sec. 2.4)
- [ ] Explicitly define `cdims_cyclic`
- [x] General: consistent usage of markdown `` vs. bold for code (e.g. compare “Computing the Frobenius norm of a tensor” to “Using reshape to rearrange elements in a tensor”)
## Creating a tenmat (tensor as matrix) object
- [ ] It would be helpful to see a full description upfront of the optional arguments rdims, cdims, and cdims_cyclic (they are not defined when calling `help(ttb.tenmat.from_tensor_type)`
- [ ] Use `rdims=`, `cdims=`, and `cdims_cyclic=` more aggressively
## Use double to convert a tenmat to a MATLAB matrix
- [x] MATLAB? Should read numpy 2D array?
## Use size and tsize for the dimensions of a tenmat
- [x] “Use shape and shape for dimensions of a tenmat”?

# Tensors
## Specifying trailing singleton dimensions in a tensor
- [x] Specify that using the `whos` commmand equivalent for obtaining details about the workspace tensors works only in interactive python environment (e.g., Jupyter Notebook, but notebooks are not mentioned in the tutorial)
## Use squeeze to remove singleton dimensions from a tensor
- [x] Provide non-trivial example:
```
np.random.seed(0) # Set a seed.
X = ttb.tenrand((2,5,4)) # Create a 3-way tensor.
np.random.seed(0) # Set the same seed.
Y = ttb.tenrand((2,5,4,1)) # Create a 4-way tensor that is the same as X but has an additional singleton dimension.
(X-Y.squeeze()).norm()
0.0
```
## Use double to convert a tensor to a (multidimensional) array
- [x] Error in comment: # Converts Y to an array of doubles. should read # Converts X to an array of doubles.
## Use ndims and size to get the size of a tensor
- [x] Replace “size” with “shape”
## Subscripted reference for a tensor
- [x] “Moreover, the subtensor is automatically renumbered/resized in the same way that MATLAB works for arrays except that singleton dimensions are handled explicitly.” Perhaps I’ve never used MATLAB? Suggest: rephrase with NumPy.
- [x] “The difference between extracting a subtensor and a list of linear indices is ambiguous for 1-dimensional tensors. We can specify ‘extract’ as a second argument whenever we are using a list of subscripts.” Where is this documented? 
    - [x] Suggestion: use `’extract’` in example.
    - [x] Consider: This may be confusing as an argument since`ktensor()` and `sptensor()` have `extract()` as class member functions. An `extract()` method for `tensor` would be consistent.
## Basic operations (plus, minus, and, or, etc.) on a tensor
- [x] Example `A / B # Calls ldivide.` appears twice.
## Using tenfun for elementwise operations on one or more tensors
- [x] Function is `tt_tenfun` not `tenfun`
## Symmetrizing and checking for symmetry in a tensor
- [x] “A second argument can also be passed to symmetrize which specifies an array of modes with respect to which the tensor should be symmetrized.” It may not be immediately obvious to the user that the “first” argument is the object itself. Suggestion: rephrase this as “An optional argument…”

**Comment by ntjohnson1** [comment]
Some non-trivial notes that will also be captured in the PR:

All the tenmat stuff will be addressed in either https://github.com/sandialabs/pyttb/issues/214 or https://github.com/sandialabs/pyttb/issues/19 (I'll copy the details over to the first). I don't currently have a great deal of understanding about the combination of arguments there and from what I remember they weren't intuitively described on the MATLAB side.

> Suggestion: re-order tutorial topics as

Won't do since it's not clear to me the value is worth the increased dev effort. Right now we just glob tutorials [here](https://github.com/sandialabs/pyttb/blob/f6e33bb30ddc19987e719b0816a7f3487e26e4af/docs/source/tutorials.rst?plain=1#L7) where the ordering would require manually listing them and remembering to update the list when a new tutorial is added.

> General: consistent usage of markdown `` vs. bold for code (e.g. compare “Computing the Frobenius norm of a tensor” to “Using reshape to rearrange elements in a tensor”)

\`code\' is the correct format as far as I'm aware so globally replaced bold for backticks in the tutorials

> Specify that using the whos commmand equivalent for obtaining details about the workspace tensors works only in interactive python environment (e.g., Jupyter Notebook, but notebooks are not mentioned in the tutorial)

IMO this isn't related to the code we write so should be removed.

> “The difference between extracting a subtensor...

In pyttb we removed the extract keyword for indexing tensors. Linear indices vs subtensor in the 1-d case should be equivalent. I just removed this note from the tutorial.
